libs/typescript/ts-utils/common-utils.ts

```diff
@@ -224,3 +225,14 @@ export function panic(message: string): never {
   console.error(message);
   return process.exit(1);
 }
+
+export function getBigIntFromLimbs(limbs: number[]): bigint {
+  let num = 0n;
+
+  for (let i = 0; i < limbs.length; i++) {
+    assert(limbs[i] <= 0xffffffff, 'Invalid limb');
+    num |= BigInt(limbs[i]) << BigInt(32 * i);
+  }
+
+  return num;
+}
```

Comment:
I think this version is slightly more clear:

```suggestion
export function getBigIntFromLimbs(words: number[]): bigint {
  let result = 0n;

  for (const word of words) {
    assert(
      Number.isInteger(word) && 0 <= word && word <= 0xffffffff,
      "Each word must fit in uint32",
    );
    result = (result << 32n) | BigInt(num);
  }

  return result;
}
```

Reply:
I am not convinced the functions are equivalent in fact I think your version is big-endian while little-endian is supposed here. I will change the assert though